### PR TITLE
doc Replace the word stage with operator in Scaladoc/Javadoc

### PR DESCRIPTION
**Pull request by richardimaoka**
Refs #24995 

Replacing the word "stage" by "operator" in Scaladoc and Javadoc comments.

Sorry for this big, discouraging diff for review... Let me know if I should reduce the amount of diff in this PR. For example, I replaced them everywhere it looks like Scaladoc or Javadoc comments, but there are internal APIs included not actually exposed to users. I can retain Scaladoc/Javadoc for internal API same as before, if needed.

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/impl/fusing/GraphInterpreterSpecKit.scala [hunk 1]
```diff
@@ -17,27 +17,27 @@ import scala.collection.{ Map ⇒ SMap }
...
+   * @param operators Operators to "materialize" into operator logic instances
+   * @param upstreams Upstream boundary logics that are already instances of operator logic and should be
+   *                  part of the graph, is placed before the rest of the operators
+   * @param downstreams Downstream boundary logics, is placed after the other operators
+   * @param attributes Optional set of attributes to pass to the operators when creating the logics
    * @return Created logics and the maps of all inlets respective outlets to those logics
    */
   private[stream] def createLogics(
-    stages:      Array[GraphStageWithMaterializedValue[_ <: Shape, _]],
+    operators:   Array[GraphStageWithMaterializedValue[_ <: Shape, _]],
```
Comment (richardimaoka) [inline akka-stream-tests/src/test/scala/akka/stream/impl/fusing/GraphInterpreterSpecKit.scala]: Replacing the parameter name, to be in line with the above Scaladoc comment change.

Comment (ktoso) [reply akka-stream-tests/src/test/scala/akka/stream/impl/fusing/GraphInterpreterSpecKit.scala]: this rename I'd skip; it really is stages here "internally"

File: akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala [hunk 2]
```diff
@@ -49,7 +49,7 @@ private[remote] trait InboundCompressions {
  * INTERNAL API
  *
  * One per incoming Aeron stream, actual compression tables are kept per-originUid and created on demand.
- * All access is via the Decoder stage.
+ * All access is via the Decoder operator.
```
Comment (ktoso) [inline akka-remote/src/main/scala/akka/remote/artery/compress/InboundCompressions.scala]: i'd keep it as stage since internal but dont mine too much

**Review by ktoso — COMMENTED** [review]
Lookin good in general, where it's internal APIs i'd probably leave it as "stage" there

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Pull request merged**